### PR TITLE
Set warnings as errors

### PR DESCRIPTION
**Pull request by Qining**
Continuation of PR#63

### CODE REVIEW

File: libshaderc/include/shaderc.h [hunk 1]
```diff
@@ -128,7 +128,10 @@ void shaderc_compile_options_set_disassembly_mode(
 void shaderc_compile_options_set_preprocessing_only_mode(
     shaderc_compile_options_t options);
 
-// Sets the compiler mode to suppress warnings.
+// Sets the compiler mode to suppress warnings, overrides warnings-as-errors
```
Comment (dekimir) [inline libshaderc/include/shaderc.h]: "overrides" -> "overriding"

File: libshaderc/src/shaderc_cpp_test.cc [hunk 2]
```diff
@@ -376,6 +413,55 @@ TEST_F(CppInterface, SuppressGlobalWarningsClonedOptions) {
...
+  // Sets the compiler to make warnings into errors. This mode should be carried
+  // into any clone of the original option object.
+  options_.SetWarningsAsErrors();
+  CompileOptions cloned_options(options_);
+  EXPECT_THAT(CompilationErrors(kMinimalUnknownVersionShader,
+                                shaderc_glsl_vertex_shader, cloned_options),
+              HasSubstr("error: version 550 is unknown.\n"));
+}
+
+TEST_F(CppInterface, SuppressWarningsModeFirstOverridesWarningsAsErrorsMode) {
```
Comment (dekimir) [inline libshaderc/src/shaderc_cpp_test.cc]: Drop "First".

File: libshaderc/src/shaderc_cpp_test.cc [hunk 3]
```diff
@@ -376,6 +413,55 @@ TEST_F(CppInterface, SuppressGlobalWarningsClonedOptions) {
...
+  // Global warnings should be inhibited.
+  EXPECT_EQ("", CompilationWarnings(kMinimalUnknownVersionShader,
+                                    shaderc_glsl_vertex_shader, options_));
+}
+
+TEST_F(CppInterface, SuppressWarningsModeSecondOverridesWarningsAsErrorsMode) {
+  // Sets warnings-as-errors mode first, then sets suppress-warnings mode.
+  // suppress-warnings mode should override warnings-as-errors mode, no
+  // error message should be output for this case.
+  options_.SetSuppressWarnings();
```
Comment (dekimir) [inline libshaderc/src/shaderc_cpp_test.cc]: This is not second. ;)

File: libshaderc/src/shaderc_test.cc [hunk 4]
```diff
@@ -323,6 +338,19 @@ TEST(CompileStringWithOptions, GlobalWarnings) {
...
+  ASSERT_NE(nullptr, compiler.get_compiler_handle());
+  const Compilation comp(compiler.get_compiler_handle(),
+                         kMinimalUnknownVersionShader,
+                         shaderc_glsl_vertex_shader, options.get());
+  EXPECT_FALSE(shaderc_module_get_success(comp.result()));
+  EXPECT_THAT(shaderc_module_get_error_message(comp.result()),
+              HasSubstr("error: version 550 is unknown.\n"));
+}
+
 TEST(CompileStringWithOptions, SuppressGlobalWarnings) {
```
Comment (dekimir) [inline libshaderc/src/shaderc_test.cc]: What about suppression cases?

Comment (Qining) [inline libshaderc/src/shaderc_test.cc]: The suppression cases are included in SuppressWarningsMode<First/Second>OverridesWarningsAsErrorsMode.
The cases with Unknown version shader.

File: libshaderc/src/shaderc_cpp_test.cc (hunk 3 above)
Comment (Qining) [inline libshaderc/src/shaderc_cpp_test.cc]: Sorry for such a fault....

File: libshaderc/src/shaderc_test.cc (hunk 4 above)
Comment (Qining) [inline libshaderc/src/shaderc_test.cc]: Oops, this is in shaderc_test.cc. I thought it is in shaderc_cpp_test.cc. I will check.

Comment (dekimir) [inline libshaderc/src/shaderc_test.cc]: Right, there are no C API tests for suppression.

**Comment by dekimir** [comment]
+2

**Comment by Qining** [comment]
merged to master #70

**Pull request closed without merging**